lib/nexthop_group.c

```diff
@@ -23,6 +23,7 @@
 #include <sockunion.h>
 #include <nexthop.h>
 #include <nexthop_group.h>
+#include <_nexthop_group.h>
```

Comment:
This naming scheme for private headers looks a bit strange to me.

What about _nexthop_group_private.h_? It's a bit on the lengthy side but the advantage is that the NG headers would appear next to each other when listing the files of the lib/ dir.

Other than that this PR looks good to me.

Reply:
Good point, fixing now